### ISSUE TITLE
Add `description` to component configuration

### ISSUE THREAD

**Issue by radu-matei** [fermyon/spin#412]
Thanks to #392, we now have the routes printed at startup:

```
Available Routes:
  http://127.0.0.1:3000 (wildcard)
  http://127.0.0.1:3000/static (wildcard)
  http://127.0.0.1:3000/favicon.ico
```

This issue proposes two things:

- also printing the component ID that corresponds to that route
- adding an optional `description` or `summary` field to the component configuration

**Comment by lianghanzhen** [comment]
This seems a good first issue, maybe I can pick it up?

**Comment by radu-matei** [comment]
That would be awesome, @lianghanzhen!

Printing the component IDs happen here:
https://github.com/fermyon/spin/blob/8bac94fceba53fbde259f0520a934e425374a198/crates/http/src/lib.rs#L220-L223

There are three main places where a description would have to be updated:

- in the manifest definition https://github.com/fermyon/spin/blob/8bac94fceba53fbde259f0520a934e425374a198/crates/manifest/src/lib.rs#L60-L70
- in the local loader
https://github.com/fermyon/spin/blob/8bac94fceba53fbde259f0520a934e425374a198/crates/loader/src/local/config.rs#L55-L73
- in the Bindle loader
https://github.com/fermyon/spin/blob/8bac94fceba53fbde259f0520a934e425374a198/crates/loader/src/bindle/config.rs#L19-L35

Let me know if I can help further.

Thanks!

**Comment by lianghanzhen** [comment]
I'm thinking about printing out the available routes in this format:
```
component id: URL (wildcard)
```

Do we need to print out the description(if applicable) as well?

**Comment by radu-matei** [comment]
Implemented in #420
